packages/twenty-front/src/pages/settings/data-model/SettingsObjectDetail.tsx

```diff
@@ -59,16 +87,91 @@ export const SettingsObjectDetail = () => {
     deactivateMetadataField,
     deleteMetadataField,
   } = useFieldMetadataItem();
+  const getRelationMetadata = useGetRelationMetadata();
+  let activeMetadataFieldsRows: MetadataFieldRowType[] = [];
```

Comment:
We should derive those states instead of using let as it breaks the immutable nature of React